tool/tsh/common/tsh.go

```diff
@@ -748,7 +743,9 @@ func Run(ctx context.Context, args []string, opts ...CliOption) error {
 
 	// run early to enable debug logging if env var is set.
 	// this makes it possible to debug early startup functionality, particularly command aliases.
-	initLogger(&cf)
+	if err := initLogger(&cf, parseDebugOptsFromEnv()); err != nil {
+		return trace.Wrap(err)
```

Comment:
I'd say failing to log shouldn't be fatal, so exiting on error seems too much. Perhaps a warning instead?

Reply:
@Tener Agree, but I'm stumped as to what to do here in case of an error. `logger.WarnContext` doesn't output anything (likely because we're yet to initialize the logger), but `slog.Default().WarnContext()` also doesn't do anything. Do I just `fmt.Printf` this?

Reply:
Huh, if `logger.WarnContext` doesn't work here then any subsequent logs will be discarded. This is rather unexpected, maybe this got run somehow?

```go
slog.SetDefault(slog.New(logutils.DiscardHandler{}))
```

Though perhaps this isn't worth the hassle to go after; so maybe add `fmt.Printf` and leave the error be?

Reply:
"Leave the error be" as in "Log the error but do not exit"? Sure.